config/rootfs/debos/overlays/chromeos-flash/opt/chromeos/flash

```diff
@@ -97,6 +97,16 @@ fixup_root()
 commands="${*:-flash_chromeos enable_rw fixup_root}"
 IMAGE_NAME="${IMAGE_NAME:-octopus-chromiumos-test-image.bin}"
 IMAGE_MOUNTPOINT="${IMAGE_MOUNTPOINT:-/root/chromeos}"
+FLASH_DEVICE="${FLASH_DEVICE:-mmcblk0}"
+
+if [ "${FLASH_DEVICE}" == "detect" ]; then
+    echo "Detecting storage device"
+    if [ -e "/dev/mmcblk0" ]; then
+       FLASH_DEVICE="mmcblk0"
+    elif [ -e "/dev/mmcblk1" ]; then
```

Comment:
Surely if `mmcblk1` exists it means there is also a `mmcblk0` and this branch is never reached.  Inverting the tests could work, but then there are other issues with doing this.

Is there any guarantee that the same device number will always map to the same device?  Typically this depends on the order they get initialised by the kernel, which may vary following random factors.  Then if they're always the same for each device type, we could pass it as a parameter in the flashing job based on some YAML config parameter.

Reply:
It will sound surprising, but thats the case with grunt: https://lava.collabora.co.uk/scheduler/job/6097834
mmcblk0 doesnt exist, but mmcblk1 does.
Worse, sometimes it books in ChromiumOS as mmcblk0, sometimes as mmcblk1, and that forced me to search alternative for tast tests root=/dev/mmcblkX as device name is not predictable (PARTLABEL should work).